### PR TITLE
Acm dev cards

### LINKED ISSUE #113: Dev Team Page Projects
Our website now has information about [ACM's Dev Team](https://www.uclaacm.com/dev) , but something else that we should have is more detail regarding the projects we do in a projects page (or add the information to the dev team page)!

Like most informational pages, creating a page requires more than just technical work - it's a team effort. This process will mimic what doing client work / working in a larger company might look like. Involved parties are:
* you, the developer(s)
* Matt + Evan + dev team director - checking the vision/copy/content
* ACM Design - helping with UI/UX, creating design elements
* everybody else on the dev team - to make sure it's representative of us!
* other people, if needed, to do user testing

The TODOS, including the ones completed in #309 are:
- [x]  what is the ACM dev team? how is it different from committee's dev teams (Hack, Teach LA), or just attending Hack events?
- [x]  who can join? how do you join? do you need previous experience?
- [x]  who runs the dev team? who's in it?
- [x]  what kind of projects do we work on? where would you find them?
- [x] what kinds of technologies do we use?
- [x]  is all of our code open-source (yes)? can anybody contribute (yes)? can you see our code (yes)?

TODOS when adding projects would be:
- [ ] can we put a list of our projects in a JSON file, and then list them out card-by-card? we can include metadata about each project; see [Teach LA's](https://teachla.uclaacm.com/dev#projects) for example - this metadata could include primary language for code, links to the project and repo in a card format instead of paragraphs like it currently is
- [ ] do we want to hit GitHub's API to show activity, or last contributions? commit graphs? insights?

Out of the box ideas include:
- [ ] any dev-related easter eggs? ex putting an application in the JS console?

The workflow should be something like:

1. identify what you want to do - how the page should be structured, what should be added and where
2. with relevant stakeholders, create a skeleton layout
3. start getting content; this can happen asynchronously, and use placeholders in the meantime
4. component-by-component, flesh out sections of the page
5. on first draft, do some user research + consult ACM Design for UI/UX pointers
6. finalize & publish 🚀 

Update:
Dev Team Page is now up ( see #309 ) 
Re-scoped issue to be project specific

This is a very general ticket with a lot of room for how you want to do it! Basically, I don't like the idea of just using paragraphs for the Dev Team projects, so I'd like to switch that into a different layout.

### PR DESCRIPTION
**Pull request by smalex-z**
# Description

ACM Dev Team Projects are now displayed as cards rather than text form. Hovering over the cards (or tapping them on mobile) shows information about the project, including the main language and the project & repo links.

Projects and Advisor cards updated as well. Project data is stored in /data/dev.js with the (now depreciated) officer data- if needed, we can update the data to be scraped from a Google Sheets like the officer data (officer-parser.mjs).

image
image

Fixes #113 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Tested across all common desktop resolutions as well as some tablet and mobile resolutions.

### CODE REVIEW

File: data/dev.js [hunk 1]
```diff
@@ -135,3 +135,69 @@ export const people = [
     alt: 'acm logo',
   },
 ];
+
+export const projdata = [
```
Comment (snigdha-kansal) [inline data/dev.js:139]: can we change this to `projData`? thanks

File: data/dev.js [hunk 2]
```diff
@@ -135,3 +135,69 @@ export const people = [
...
+    repo_link: 'https://github.com/uclaacm/membership-portal',
+    description: `The Membership Portal is where ACM members can check in to events to earn 
+    points! We built the Membership Portal with Node.js, Express.js, PostgreSQL, Redux and React.js.`,
+    img: '/images/dev-projects/membership.png',
+    alt: 'Membership Portal!',
+  },{
+    name: 'Open Source',
+    prim_lang: 'JS',
+    proj_link: 'https://opensource.uclaacm.com/',
+    repo_link: 'https://github.com/uclaacm',
```
Comment (snigdha-kansal) [inline data/dev.js:162]: repo_link for open source project is 'https://github.com/uclaacm/opensource'

File: data/dev.js [hunk 3]
```diff
@@ -135,3 +135,69 @@ export const people = [
...
+    description: `Everything we build is open source! In the spirit of open-source, anyone is welcome 
+    to view and contribute to our projects on Github.`,
+    img: '/images/dev-projects/opensource.png',
+    alt: 'Open Source!',
+  },{
+    name: 'CMS Template',
+    prim_lang: 'JS',
+    proj_link: 'https://github.com/uclaacm/committee-website-template-cms',
+    repo_link: 'https://github.com/uclaacm/committee-website-template-cms',
+    description: 'This Committee Website template enables any ACM Committee to put up their website in minutes!',
```
Comment (snigdha-kansal) [inline data/dev.js:172]: can we change description to:
This Committee Website template is a no-code solution that empowers any ACM committee to build and deploy their website in minutes! It's actively in use by multiple committees as it saves time and ensures consistency across sites.

File: data/dev.js [hunk 4]
```diff
@@ -135,3 +135,69 @@ export const people = [
...
+    description: `The Membership Portal is where ACM members can check in to events to earn 
+    points! We built the Membership Portal with Node.js, Express.js, PostgreSQL, Redux and React.js.`,
+    img: '/images/dev-projects/membership.png',
+    alt: 'Membership Portal!',
+  },{
+    name: 'Open Source',
+    prim_lang: 'JS',
+    proj_link: 'https://opensource.uclaacm.com/',
+    repo_link: 'https://github.com/uclaacm/opensource',
+    description: `Everything we build is open source! In the spirit of open-source, anyone is welcome
```
Comment (snigdha-kansal) [inline data/dev.js:163]: can we change description to:
Open Source highlights all open source projects managed by [ACM at UCLA](https://github.com/uclaacm), accompanied by an activity feed showcasing recent contributions to [repositories under ACM at UCLA](https://github.com/orgs/uclaacm/repositories). Anyone is welcome to contribute, making it an open invitation to get involved and make an impact!

File: pages/dev.js [hunk 5]
```diff
@@ -2,25 +2,26 @@ import { NextSeo } from 'next-seo';
 import Link from 'next/link';
 import React from 'react';
 
+import Officers from '../components/OfficerCard';
```
Comment (snigdha-kansal) [inline pages/dev.js:5]: can you fix the lint error for this? thanks

File: pages/dev.js [hunk 6]
```diff
@@ -2,25 +2,26 @@ import { NextSeo } from 'next-seo';
...
+import Projects from '../components/DevProjectCards';
```
Comment (snigdha-kansal) [inline pages/dev.js:6]: can you fix the lint error for this? thanks

File: data/dev.js [hunk 7]
```diff
@@ -135,3 +135,69 @@ export const people = [
...
+    description: `The link shortener is an ongoing project aimed at generating shortened links
+     for use by our committees. The link shortener is built with Flask.`,
+    img: '/images/dev-projects/acmlogocopy.jpg',
+    alt: 'ACM Logo!',
+  },{
+    name: '???',
+    prim_lang: 'Any',
+    proj_link: 'mailto:arshmalik02@g.ucla.edu',
+    repo_link: 'mailto:arshmalik02@g.ucla.edu',
+    description: `Have an idea for a project or any questions? Feel free to contact Arsh Malik
```
Comment (snigdha-kansal) [inline data/dev.js:198]: can we add a newline between the sentences?

File: data/dev.js [hunk 8]
```diff
@@ -135,3 +135,69 @@ export const people = [
...
+    prim_lang: 'Flask',
+    proj_link: 'http://links.uclaacm.com/',
+    repo_link: '',
+    description: `The link shortener is an ongoing project aimed at generating shortened links
+     for use by our committees. The link shortener is built with Flask.`,
+    img: '/images/dev-projects/acmlogocopy.jpg',
+    alt: 'ACM Logo!',
+  },{
+    name: '???',
+    prim_lang: 'Any',
```
Comment (snigdha-kansal) [inline data/dev.js:195]: can we change to '???'

Comment (snigdha-kansal) [reply data/dev.js:195]: also if we can just render '???' and omit the rest that follows?

File: data/dev.js [hunk 9]
```diff
@@ -135,3 +135,69 @@ export const people = [
...
+    proj_link: 'mailto:arshmalik02@g.ucla.edu',
+    repo_link: 'mailto:arshmalik02@g.ucla.edu',
+    description: `Have an idea for a project or any questions? Feel free to contact Arsh Malik 
+    (arshmalik02@g.ucla.edu), our Dev Team Director.`,
```
Comment (snigdha-kansal) [inline data/dev.js:199]: hyper link the email here?
<a href="mailto:arshmalik02@g.ucla.edu">arshmalik02@g.ucla.edu</a>

File: styles/components/DevProjCard.module.scss [hunk 10]
```diff
@@ -0,0 +1,245 @@
...
+  padding-bottom: 0;
+}
+
+.card-hover-content {
+  align-items: center;
+  background-color: rgba(255, 255, 255, 0.9);
+  color: black;
+  display: flex;
+  flex-direction: column;
+  font-family: sans-serif;
```
Comment (snigdha-kansal) [inline styles/components/DevProjCard.module.scss:42]: change to `$font-body`? as per https://design.uclaacm.com/guidelines/

File: data/dev.js (hunk 4 above)
Comment (smalex-z) [reply data/dev.js:163]: Changed a bit to fit in the conventional space

**Comment by smalex-z** [comment]
CSS has been updated to make everything more clear and viewable. Additionally grid view for tablets should look better now with some JS and CSS changes. Description box for projects supports in-line HTML too now. Try to keep each line to 100 characters max for readability

**Review by snigdha-kansal — APPROVED** [review]
GIF

Looks great, Alex 💯 
Thanks so much for doing this!!

Note: The font size change is a bit abrupt (goes from very large to very small) around 767-768px -- something to fix later.

**Pull request merged**